File: src/main/java/com/lalkalol/mujahid/util/Embeds.java
package com.lalkalol.mujahid.util;

import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.entities.MessageEmbed;

import java.awt.Color;

public final class Embeds {
    public static final Color BRAND = new Color(0x5865F2);
    public static final Color SUCCESS = new Color(0x57F287);
    public static final Color WARNING = new Color(0xFEE75C);
    public static final Color ERROR = new Color(0xED4245);
    public static final Color MUSIC = new Color(0xEB459E);

    private Embeds() {
    }

    public static MessageEmbed success(String text) {
        return simple(SUCCESS, text);
    }

    public static MessageEmbed warning(String text) {
        return simple(WARNING, text);
    }

    public static MessageEmbed error(String text) {
        return simple(ERROR, text);
    }

    public static MessageEmbed info(String text) {
        return simple(BRAND, text);
    }

    public static EmbedBuilder music() {
        return new EmbedBuilder().setColor(MUSIC);
    }

    private static MessageEmbed simple(Color color, String description) {
        return new EmbedBuilder()
                .setColor(color)
                .setDescription(description)
                .build();
    }
}

File: src/main/kotlin/com/lalkalol/mujahid/util/Embeds.kt
package com.lalkalol.mujahid.util

import net.dv8tion.jda.api.EmbedBuilder
import net.dv8tion.jda.api.entities.MessageEmbed
import java.awt.Color

object Embeds {
    val BRAND: Color = Color(0x5865F2)
    val SUCCESS: Color = Color(0x57F287)
    val WARNING: Color = Color(0xFEE75C)
    val ERROR: Color = Color(0xED4245)
    val MUSIC: Color = Color(0xEB459E)

    fun success(text: String): MessageEmbed = simple(SUCCESS, text)

    fun warning(text: String): MessageEmbed = simple(WARNING, text)

    fun error(text: String): MessageEmbed = simple(ERROR, text)

    fun info(text: String): MessageEmbed = simple(BRAND, text)

    fun music(): EmbedBuilder = EmbedBuilder().setColor(MUSIC)

    private fun simple(color: Color, description: String): MessageEmbed =
        EmbedBuilder()
            .setColor(color)
            .setDescription(description)
            .build()
}
